Demo/Demo/RootWindow.swift

```diff
@@ -1,12 +1,23 @@
 import UIKit
+import Combine
 import Components
 
 /// The root window for the demo app. Trait collection changes are fed into the Components system here.
 final class RootWindow: UIWindow {
-    
+
+	// MARK: - Private Properties
+
+	private var cancellables = Set<AnyCancellable>()
+
+	// MARK: - Lifecycle
+
     override init(windowScene: UIWindowScene) {
         super.init(windowScene: windowScene)
-        
+
+		WKAppEnvironment.current.$theme.sink(receiveValue: { theme in
+			self.overrideUserInterfaceStyle = theme.userInterfaceStyle
```

Comment:
I noticed that with this, the Default theme no longer automatically switches themes when the system mode changes. It's a little gross, but changing the assignment to this seems to fix it:

```
self.overrideUserInterfaceStyle = UserDefaults.standard.themeName == "Default" ? .unspecified : theme.userInterfaceStyle
```

Let me know what you think!

Reply:
Nice catch! Updated with your suggestion. I went down a little rabbit and noticed some inconsistent behavior updating views when the theme in the demo app is updated, but I'll save that for later because it's not explicitly related to the Components framework itself or this button work, just the demo app.